### ISSUE TITLE
Comparison of the results of different versions of pymoo

### ISSUE THREAD

**Issue by Zhenyu2Liang** [EMI-Group/evoxbench#6]
We solve test instances in **EvoXBench (v1.0.3)** using algorithms from different versions of the pymoo platform **(pymoo v0.6.0 and pymoo v0.5.0).** The algorithms we use are NSGAII, MOEA/D and NSGAIII. All experiments were carried out under the same conditions as the paper. The following are the statistical results. 

 

TABLE I: Statistical results (median and standard deviation) of the HV values on C-10/MOP test suite using **pymoo 0.6.0**. The best results of each instance are in bold.

|           |        NSGAII        |      MOEA/D       |       NSGAIII        |
| :-------: | :------------------: | :--------------: | :------------------: |
| C-10/MOP1 |   0.9191 (0.0088)=   | 0.8852 (0.0244)- | **0.9193 (0.0066)+** |
| C-10/MOP2 | **0.8893 (0.0083)+** | 0.8638 (0.0078)- |   0.3964 (0.2444)-   |
| C-10/MOP3 | **0.8252 (0.0013)+** | 0.7933 (0.0076)- |   0.8065 (0.0025)-   |
| C-10/MOP4 | **0.7761 (0.0058)+** | 0.7284 (0.0144)- |   0.7641 (0.0060)-   |
| C-10/MOP5 | **0.7125 (0.0001)+** | 0.6705 (0.0331)- |   0.7032 (0.0139)-   |
| C-10/MOP6 | **0.7374 (0.0029)+** | 0.6794 (0.0146)- |   0.6616 (0.0218)-   |
| C-10/MOP7 | **0.5705 (0.0116)+** | 0.5100 (0.0808)- |   0.5394 (0.0323)=   |
| C-10/MOP8 | **0.9738 (0.0021)+** | 0.8889 (0.0135)- |   0.9684 (0.0027)-   |
| C-10/MOP9 | **0.9561 (0.0019)+** | 0.6820 (0.0615)- |   0.9492 (0.0023)-   |

\+ indicates a method achieving significantly better performance. 

= indicates a method achieving similar performance as the best-performing method. 

− indicates a method achieving significantly worse performance.

TABLE II: Statistical results (median and standard deviation) of the HV values on C-10/MOP test suite using **pymoo 0.5.0**. The best results of each instance are in bold.

|           |        NSGAII        |      MOEA/D      |     NSGAIII      |
| :-------: | :------------------: | :--------------: | :--------------: |
| C-10/MOP1 | **0.9386 (0.0047)+** | 0.9131 (0.0164)- | 0.9336 (0.0041)= |
| C-10/MOP2 | **0.9176 (0.0028)+** | 0.8733 (0.0063)- | 0.6746 (0.2402)- |
| C-10/MOP3 | **0.8243 (0.0020)+** | 0.8006 (0.0047)- | 0.8080 (0.0037)- |
| C-10/MOP4 | **0.7643 (0.0069)+** | 0.7338 (0.0163)- | 0.7612 (0.0075)= |
| C-10/MOP5 | **0.7127 (0.0001)+** | 0.6729 (0.0067)- | 0.7031 (0.0167)- |
| C-10/MOP6 | **0.7404 (0.0001)+** | 0.6805 (0.0652)- | 0.6387 (0.0146)- |
| C-10/MOP7 | **0.5745 (0.0121)+** | 0.5149 (0.0691)- | 0.5364 (0.0181)- |
| C-10/MOP8 | **0.9763 (0.0061)+** | 0.8906 (0.0193)- | 0.9740 (0.0060)- |
| C-10/MOP9 | **0.9638 (0.0069)+** | 0.7812 (0.0394)- | 0.9568 (0.0131)- |

\+ indicates a method achieving significantly better performance. 

= indicates a method achieving similar performance as the best-performing method. 

− indicates a method achieving significantly worse performance.

In NSGAII, pymoo 0.6.0 shows better performance on C-10/MOP3, C-10/MOP4. pymoo 0.5.0 performs better except for these test instances. pymoo 0.6.0 made changes to sampling, crossover, and mutation operators, which may have unexpected effects on these test instances. However, the impact of version changes on performance is relatively small, and the maximum difference of HV does not exceed 0.0283. It is recommended to use the latest version of pymoo. Users who participate in the competition can choose to use two versions according to their needs.

**Comment by Strepsicrates** [comment]
I can not agree with your comment. You only pay attention to the HV value change of NSGA-II. In fact, the impact of version changes on performance is very large. As for NSGA-III, the maximum difference of HV exceed 0.2782 on C-10/MOP2.  MOEA/D also has a similar situation on C-10/MOP9. In addition, the performance of different algorithms on the IN-1K/MOP test suite will also be affected by the Pymoo version.  Therefore, all submitted results should use the same version(latest version) of Pymoo and EvoXBench to calculate HV and IGD, otherwise, the NAS competition is not fair.

**Comment by mikelzc1990** [comment]
Thank you for bringing up the issue of variations in performance metrics due to different versions of Pymoo for certain EMO algorithms. While we appreciate your concern, we would like to clarify that the EvoXBench benchmark test suites, just like any other EMO benchmark, do not depend on any specific EMO framework, including Pymoo. As such, the use of different versions of Pymoo or any other EMO algorithm framework should not affect the fairness of the NAS competition.

We acknowledge that some users or competition participants may prefer to use the latest version of Pymoo or other EMO algorithm frameworks, and we encourage them to do so. However, we believe that it is ultimately up to the user or participant to decide which EMO algorithm framework to use.

In summary, we thank you for your comment and appreciate your concern for the fairness of the NAS competition. However, we respectfully disagree with your statement that the use of different versions of Pymoo would render the competition unfair. If you have any further questions or concerns, please do not hesitate to let us know.
